### PR TITLE
API CreateSalaryChangeRequestFromCheckpointTool

### PR DESCRIPTION
**Pull request by quangnguyenhongncc**
Make API CreateSalaryChangeRequestFromCheckpointTool

### CODE REVIEW

File: aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs [hunk 1]
```diff
@@ -377,8 +377,74 @@ public async Task<GridResult<GetRequestEmployeeDto>> GetEmployeesInSalaryRequest
...
+            var listResult = new List<string>();
+            var listRequestChangeSalary = new List<SalaryChangeRequestEmployee>();
+            long employeeId;
+            double realSalary;
+
+            foreach (var dto in input.RequestChangeSalaryEmployees)
+            {
+                try
+                {
+                    long newLevelId = dictLevel[dto.ToLevelCode.ToLower().Trim()].Id;
```
Comment (tiennguyenhuuncc) [inline aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs]: ko kiểm tra tồn tại key trong dicLevel à?

Comment (quangnguyenhongncc) [reply aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs]: đã update ạ

File: aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs [hunk 2]
```diff
@@ -377,8 +377,79 @@ public async Task<GridResult<GetRequestEmployeeDto>> GetEmployeesInSalaryRequest
...
+            var listResult = new List<string>();
+            var listRequestChangeSalary = new List<SalaryChangeRequestEmployee>();
+            long employeeId;
+            double realSalary;
+
+            foreach (var dto in input.RequestChangeSalaryEmployees)
+            {
+                try
+                {
+                    if (!dictLevel.ContainsKey(dto.ToLevelCodeToLowerTrim))
```
Comment (tiennguyenhuuncc) [inline aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs:431]: if (!dictLevel.ContainsKey(dto.ToLevelCodeToLowerTrim))
                    {
                        listResult.Add($"{dto.EmailAddress} - {dto.ToLevelCode} - fail: not found level code in HRM");
                        continue;
                    }

File: aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs [hunk 3]
```diff
@@ -377,8 +377,79 @@ public async Task<GridResult<GetRequestEmployeeDto>> GetEmployeesInSalaryRequest
...
+                    }
+                    if (!dicEmployee.ContainsKey(dto.EmailAddressToLowerTrim)) 
+                    {
+                        listResult.Add($"{dto.EmailAddress} - fail: not found in HRM");
+                        continue;
+                    }
+                    long newLevelId = dictLevel[dto.ToLevelCodeToLowerTrim].Id;
+                    employeeId = dicEmployee[dto.EmailAddressToLowerTrim].Id;
+                    realSalary = dicEmployee[dto.EmailAddressToLowerTrim].RealSalary;
+                    await WorkScope.InsertAsync(new SalaryChangeRequestEmployee()
```
Comment (tiennguyenhuuncc) [inline aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs:442]: entity SalaryChangeRequestEmployee có nhiều field lắm, sao truyền vào ít thế này!

File: aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs [hunk 4]
```diff
@@ -377,8 +377,81 @@ public async Task<GridResult<GetRequestEmployeeDto>> GetEmployeesInSalaryRequest
...
+              })
+              .ToList()
+              .GroupBy(x => x.Email.ToLower().Trim())
+              .ToDictionary(x => x.Key, s => s.FirstOrDefault());
+
+            var listResult = new List<string>();
+            var listRequestChangeSalary = new List<SalaryChangeRequestEmployee>();
+            double realSalary;
+
+            foreach (var dto in input.RequestChangeSalaryEmployees)
```
Comment (dattranthanhncc) [inline aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs:427]: rename dto -> requestChangeSalaryEmployee

File: aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs [hunk 5]
```diff
@@ -377,8 +377,81 @@ public async Task<GridResult<GetRequestEmployeeDto>> GetEmployeesInSalaryRequest
...
+                        continue;
+                    }
+                    if (!dicEmployee.ContainsKey(dto.EmailAddressToLowerTrim)) 
+                    {
+                        listResult.Add($"{dto.EmailAddress} - fail: not found in HRM");
+                        continue;
+                    }
+                    long newLevelId = dictLevel[dto.ToLevelCodeToLowerTrim].Id;
+                    var employee = dicEmployee[dto.EmailAddressToLowerTrim];
+                    realSalary = dicEmployee[dto.EmailAddressToLowerTrim].RealSalary;
```
Comment (tiennguyenhuuncc) [inline aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs]: remove realSalary = dicEmployee[dto.EmailAddressToLowerTrim].RealSalary;

File: aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs [hunk 6]
```diff
@@ -377,8 +377,81 @@ public async Task<GridResult<GetRequestEmployeeDto>> GetEmployeesInSalaryRequest
...
+                    await WorkScope.InsertAsync(new SalaryChangeRequestEmployee()
+                    {
+                        EmployeeId = employee.Id,
+                        LevelId = employee.Level.Id, 
+                        ToLevelId = newLevelId,
+                        FromUserType = employee.UserType,
+                        ToUserType = employee.UserType,
+                        Salary = employee.Salary, 
+                        ToSalary = dto.SalaryIncrease + realSalary,
```
Comment (tiennguyenhuuncc) [inline aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs]: ToSalary = dto.SalaryIncrease + realSalary,
=> ToSalary = dto.SalaryIncrease + employee.RealSalary,

File: aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs [hunk 7]
```diff
@@ -377,8 +377,81 @@ public async Task<GridResult<GetRequestEmployeeDto>> GetEmployeesInSalaryRequest
...
+                    {
+                        listResult.Add($"{dto.EmailAddress} - {dto.ToLevelCode} - fail: not found level code in HRM");
+                        continue;
+                    }
+                    if (!dicEmployee.ContainsKey(dto.EmailAddressToLowerTrim)) 
+                    {
+                        listResult.Add($"{dto.EmailAddress} - fail: not found in HRM");
+                        continue;
+                    }
+                    long newLevelId = dictLevel[dto.ToLevelCodeToLowerTrim].Id;
```
Comment (dattranthanhncc) [inline aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs]: remove variable newLevelId

File: aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs [hunk 8]
```diff
@@ -377,8 +377,81 @@ public async Task<GridResult<GetRequestEmployeeDto>> GetEmployeesInSalaryRequest
...
+                    var employee = dicEmployee[dto.EmailAddressToLowerTrim];
+                    realSalary = dicEmployee[dto.EmailAddressToLowerTrim].RealSalary;
+                    await WorkScope.InsertAsync(new SalaryChangeRequestEmployee()
+                    {
+                        EmployeeId = employee.Id,
+                        LevelId = employee.Level.Id, 
+                        ToLevelId = newLevelId,
```
Comment (dattranthanhncc) [inline aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/SalaryRequestManager.cs]: ToLevelId = dictLevel[dto.ToLevelCodeToLowerTrim].Id

**Review by tiennguyenhuuncc — APPROVED** [review]

**Pull request merged**